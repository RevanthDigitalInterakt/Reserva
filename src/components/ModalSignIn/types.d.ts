import type { Dispatch, SetStateAction } from 'react';

export type IParamsComponent = {
  onClose(): void;
  isVisible: boolean;
  loadingAddCart: boolean;
  setAnimationBag: Dispatch<SetStateAction<boolean>>;
  setLoadingAddCart: Dispatch<SetStateAction<boolean>>;
};
