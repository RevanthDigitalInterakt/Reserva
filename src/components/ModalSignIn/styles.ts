import { StyleSheet } from 'react-native';
import configDeviceSizes from '../../utils/configDeviceSizes';

export const objectStyles = StyleSheet.create({
  modal: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  containerModal: {
    width: configDeviceSizes.DEVICE_WIDTH - 48,
    top: -60,
  },
  wrapperAboutPrime: {
    marginTop: 8,
    marginBottom: 24,
  },
  footerDescription: {
    marginTop: 12,
  },
  footerHighlight: {
    textDecorationLine: 'underline',
  },
  headerDescription: {
    marginTop: 24,
    marginBottom: 24,
  },
});
