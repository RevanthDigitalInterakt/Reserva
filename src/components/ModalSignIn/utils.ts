import * as Yup from 'yup';

export const isValidPassword = (text: string) => Yup.string()
  .required()
  .matches(/^(?=.{8,})/) // 8 caracteres
  .matches(/^(?=.*[A-Z])/) // pelo menos uma maiuscula
  .matches(/^(?=.*[a-z])/) // pelo menos uma minuscula
  .matches(/^(?=.*[0-9])/) // pelo menos um nuemro
  .isValidSync(text);

export const isValidEmail = (text: string) => Yup.string()
  .required().email().isValidSync(text.trim());
