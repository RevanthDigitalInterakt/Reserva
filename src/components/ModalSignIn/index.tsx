import React from 'react';
import Modal from 'react-native-modal';
import { ScrollView, TouchableOpacity, View } from 'react-native';
import { Box, Icon, Typography } from '@usereservaapp/reserva-ui';
import { useNavigation } from '@react-navigation/native';

import testProps from '../../utils/testProps';
import EventProvider from '../../utils/EventProvider';
import { useAuthentication } from '../../hooks/useAuthentication';
import { useAuthStore } from '../../zustand/useAuth/useAuthStore';

import { Button } from '../Button';
import UnderlineInput from '../UnderlineInput';
import { FooterModalPrime } from './components/FooterModal';

import * as Styles from './styles';
import type { IParamsComponent } from './types';
import { isValidEmail, isValidPassword } from './utils';
import IconLogoPrime from '../../../assets/icons/IconLogoPrime';

export const ModalSignIn: React.FC<IParamsComponent> = ({
  onClose,
  isVisible,
  onModalHide,
}) => {
  const { navigate } = useNavigation();
  const { profile } = useAuthStore(['profile']);

  const {
    loadingSignIn,
    handleLogin,
    loginCredentials,
    setLoginCredentials,
    setEmailIsValid,
    setPasswordIsValid,
  } = useAuthentication({ closeModal: onClose });

  return (
    <Modal
      animationIn="fadeIn"
      animationOut="zoomOut"
      isVisible={isVisible}
      animationInTiming={300}
      onBackdropPress={onClose}
      onModalHide={onModalHide}
      style={Styles.objectStyles.modal}
      {...testProps('com.usereserva:id/modal_sign_in')}
    >
      <Box
        p="xxxs"
        backgroundColor="white"
        style={Styles.objectStyles.containerModal}
        {...testProps('com.usereserva:id/modal_sign_in_container')}
      >
        <Box flexDirection="row-reverse">
          <TouchableOpacity
            onPress={onClose}
            {...testProps('com.usereserva:id/modal_sign_in_close')}
          >
            <Icon name="Close" size={14} />
          </TouchableOpacity>
        </Box>

        <IconLogoPrime
          {...testProps('com.usereserva:id/modal_sign_in_title')}
        />

        <ScrollView showsVerticalScrollIndicator={false}>
          {!profile?.email && (
            <>
              <Box style={Styles.objectStyles.headerDescription}>
                <Typography fontSize={14} color="neutroFrio2">
                  Olá! Já é cliente Prime? Então basta inserir seu e-mail abaixo
                  para poder comprar:
                </Typography>
              </Box>

              <Box>
                <UnderlineInput
                  isSecureText={false}
                  keyboardType="email-address"
                  placeholder="Digite seu e-mail"
                  value={loginCredentials.username}
                  errorMsg={loginCredentials.usernameError}
                  testID="com.usereserva:id/modal_sign_in_input_email"
                  showError={loginCredentials.showUsernameError}
                  onChangeText={(text) => {
                    try {
                      setLoginCredentials({
                        ...loginCredentials,
                        username: text,
                      });
                      setEmailIsValid(isValidEmail(text));
                    } catch (error) {
                      EventProvider.sentry.captureException(error, {
                        extra: {
                          writtenEmail: text,
                        },
                      });
                    }
                  }}
                />

                <View style={{
                  marginTop: 24,
                }}
                >
                  <UnderlineInput
                    isSecureText
                    placeholder="Digite sua senha"
                    value={loginCredentials.password}
                    showError={loginCredentials.showPasswordError}
                    testID="com.usereserva:id/modal_sign_in_input_password"
                    onChangeText={(text) => {
                      setLoginCredentials({
                        ...loginCredentials,
                        password: text,
                      });
                      setPasswordIsValid(isValidPassword(text));
                    }}
                  />
                  <Box mt="micro" mb="quarck">
                    <TouchableOpacity
                      {...testProps(
                        'com.usereserva:id/modal_sign_in_cta_forgot_password',
                      )}
                      onPress={() => {
                        onClose();
                        navigate('ForgotEmail', {});
                      }}
                    >
                      <Typography
                        fontSize={14}
                        style={{ textDecorationLine: 'underline' }}
                      >
                        Esqueci minha senha
                      </Typography>
                    </TouchableOpacity>
                  </Box>

                  {loginCredentials.hasError && (
                    <Typography
                      fontSize={13}
                      color="vermelhoAlerta"
                      fontFamily="nunitoRegular"
                    >
                      {loginCredentials.showMessageError}
                    </Typography>
                  )}
                </View>
              </Box>

              <Button
                inline
                mt="xs"
                mb="xxs"
                onPress={handleLogin}
                disabled={loadingSignIn}
                variant="primarioEstreito"
                title="CONTINUAR COMPRANDO"
                testID="com.usereserva:id/modal_sign_in-cta_enter"
              />

              <Box
                alignItems="center"
                flexDirection="row"
                justifyContent="center"
                style={Styles.objectStyles.wrapperAboutPrime}
              >
                <Box
                  flex={1}
                  marginRight="md"
                  borderColor="divider"
                  borderWidth="hairline"
                />
                <Typography fontSize={14} textAlign="center">
                  Ainda não é
                  {' '}
                  <Typography fontSize={14} fontFamily="reservaSerifBlack">
                    Prime
                  </Typography>
                  ?
                </Typography>
                <Box
                  flex={1}
                  marginLeft="md"
                  borderColor="divider"
                  borderWidth="hairline"
                />
              </Box>
            </>
          )}

          <FooterModalPrime onClose={onClose} />
        </ScrollView>
      </Box>
    </Modal>
  );
};
