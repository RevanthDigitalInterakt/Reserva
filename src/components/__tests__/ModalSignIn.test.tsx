import React from 'react';
import {
  render,
  fireEvent,
} from '@testing-library/react-native';
import { act } from '@testing-library/react-hooks';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from 'styled-components/native';

import { CartContext } from '../../context/CartContext';
import { mockPrimeData } from '../../../__mocks__/PrimeLP.mock';
import * as useLandingPagePrimeQuery from '../../base/graphql/generated';
import { ApolloMockLPPrime } from '../../pages/PrimeLP/__mocks__/primeLPMocks';

import { ModalSignIn } from '../ModalSignIn';
import { isValidEmail, isValidPassword } from '../ModalSignIn/utils';
import { theme } from '../../base/usereservappLegacy/theme';

const mockAddItemFn = jest.fn();
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockedNavigate }),
}));

jest.mock('../../hooks/usePrimeInfo', () => ({
  usePrimeInfo: () => ({
    onAddPrimeToCart: mockAddItemFn,
  }),
}));

jest.mock('../../zustand/useDitoStore', () => ({
  useDitoStore: () => ({
  }),
}));

jest.mock('../../zustand/useApolloFetchPolicyStore', () => ({
  useApolloFetchPolicyStore: () => ({
    getFetchPolicyPerKey: () => 'network-only',
  }),
}));

jest
  .spyOn(useLandingPagePrimeQuery, 'useLandingPagePrimeQuery')
  .mockReturnValue({
    data: {
      landingPagePrime: {
        ...mockPrimeData,
      },
    },
    loading: false,
  } as any);

const Component = (
  <ThemeProvider theme={theme}>
    <MockedProvider mocks={ApolloMockLPPrime} addTypename={false}>
      <CartContext.Provider value={
      { addItem: mockAddItemFn } as any
    }
      >
        <ModalSignIn
          onClose={jest.fn()}
          isVisible
        />
      </CartContext.Provider>
    </MockedProvider>
  </ThemeProvider>
);

describe('ModalSignIn', () => {
  it('should match with the snapshot', async () => {
    const { toJSON } = render(Component);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render element correctly', async () => {
    const { getByTestId } = render(Component);

    const elementWrapper = getByTestId('com.usereserva:id/modal_sign_in');

    expect(elementWrapper).toBeTruthy();
  });

  describe('should be able correctly call in buttons modal', () => {
    it('Add Prime to cart', async () => {
      const { getByTestId } = render(Component);

      const addPrimeToCart = getByTestId('com.usereserva:id/modal_sign_in_cta_add_prime');

      await act(async () => {
        await fireEvent.press(addPrimeToCart);
      });

      expect(mockAddItemFn).toHaveBeenCalled();
    });

    it('Forgot password', async () => {
      const { getByTestId } = render(Component);

      const goToForgotPassword = getByTestId('com.usereserva:id/modal_sign_in_cta_forgot_password');

      await act(async () => {
        await fireEvent.press(goToForgotPassword);
      });

      expect(mockedNavigate).toHaveBeenCalledWith('ForgotEmail', {});
    });

    it('Redirect to Landing Page Prime', async () => {
      const { getByTestId } = render(Component);

      const clickHereRedirectToLPPrime = getByTestId('com.usereserva:id/modal_sign_click_here');

      await act(async () => {
        await fireEvent.press(clickHereRedirectToLPPrime);
      });

      expect(mockedNavigate).toHaveBeenCalledWith('PrimeLP');
    });
  });

  describe('should be able correctly change texts in fields', () => {
    it('Input E-mail', async () => {
      const { getByTestId } = render(Component);

      const inputEmail = getByTestId('com.usereserva:id/modal_sign_in_input_email');

      await act(async () => {
        await fireEvent.changeText(inputEmail, 'example@example.com');
      });

      expect(inputEmail.props.value).toBe('example@example.com');
    });

    it('Input Password', async () => {
      const { getByTestId } = render(Component);

      const inputPassword = getByTestId('com.usereserva:id/modal_sign_in_input_password');

      await act(async () => {
        await fireEvent.changeText(inputPassword, 'Example@123');
      });

      expect(inputPassword.props.value).toBe('Example@123');
    });
  });

  describe('Utils', () => {
    describe('isValidPassword', () => {
      it('returns true for a valid password', () => {
        const validPassword = 'Abcdefg1';
        const isValid = isValidPassword(validPassword);
        expect(isValid).toBe(true);
      });

      it('returns false for an invalid password', () => {
        const invalidPassword = 'password';
        const isValid = isValidPassword(invalidPassword);
        expect(isValid).toBe(false);
      });
    });

    describe('isValidEmail', () => {
      it('returns true for a valid email', () => {
        const validEmail = 'test@example.com';
        const isValid = isValidEmail(validEmail);
        expect(isValid).toBe(true);
      });

      it('returns false for an invalid email', () => {
        const invalidEmail = 'invalidemail';
        const isValid = isValidEmail(invalidEmail);
        expect(isValid).toBe(false);
      });
    });
  });
});
