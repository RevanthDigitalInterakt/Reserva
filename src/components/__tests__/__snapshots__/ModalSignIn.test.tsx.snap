// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSignIn should match with the snapshot 1`] = `
<Modal
  accessibilityLabel="com.usereserva:id/modal_sign_in"
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="com.usereserva:id/modal_sign_in"
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    accessibilityLabel="com.usereserva:id/modal_sign_in"
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      accessibilityLabel="com.usereserva:id/modal_sign_in_container"
      backgroundColor="white"
      height={1208}
      p="xxxs"
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
            "height": 1208,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 726,
          },
        ]
      }
      testID="com.usereserva:id/modal_sign_in_container"
      width={726}
    >
      <View
        flexDirection="row-reverse"
        style={
          Array [
            Object {
              "flexDirection": "row-reverse",
            },
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/modal_sign_in_close"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="com.usereserva:id/modal_sign_in_close"
        >
          <RNSVGSvgView
            bbHeight={14}
            bbWidth={14}
            color="preto"
            focusable={false}
            size={14}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "color": "#333333",
                    "height": 14,
                    "width": 14,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 14,
                  "width": 14,
                },
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={8.24}
                vbWidth={8.242}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M5.098 4.12l2.944-2.944A.69.69 0 007.066.2L4.122 3.144 1.178.2a.69.69 0 10-.976.976L3.146 4.12.202 7.064a.69.69 0 00.976.976l2.944-2.944L7.066 8.04a.69.69 0 00.976-.976z"
                    fill={4281545523}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <Text
        fontFamily="reservaSerifBlack"
        style={
          Array [
            Object {
              "fontFamily": "ReservaSerif-Bold",
              "fontSize": 27,
            },
          ]
        }
        variant="descontoTag1"
      >
        Prime
      </Text>
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View
            my="micro"
            style={
              Array [
                Object {
                  "marginBottom": 12,
                  "marginTop": 12,
                },
              ]
            }
          >
            <Text
              color="neutroFrio2"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#8A8C8E",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Olá! Já é cliente Prime? Então basta inserir seu e-mail abaixo para poder comprar:
            </Text>
          </View>
          <View
            mt="xxs"
            style={
              Array [
                Object {
                  "marginTop": 24,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "width": 710,
                  },
                ]
              }
              width={710}
            >
              <View
                borderBottomColor="neutroFrio2"
                borderBottomWidth="hairline"
                flexDirection="row"
                justifyContent="space-between"
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#8A8C8E",
                      "borderBottomWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                    Object {
                      "overflow": "hidden",
                    },
                  ]
                }
              >
                <View
                  flexGrow={4}
                  style={
                    Array [
                      Object {
                        "flexGrow": 4,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="com.usereserva:id/modal_sign_in_input_email"
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={true}
                    collapsable={false}
                    keyboardType="email-address"
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onGestureHandlerEvent={[Function]}
                    onGestureHandlerStateChange={[Function]}
                    placeholder="Digite seu e-mail"
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Object {
                        "margin": 0,
                        "maxWidth": 710,
                        "padding": 0,
                      }
                    }
                    testID="com.usereserva:id/modal_sign_in_input_email"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              mt="md"
              style={
                Array [
                  Object {
                    "marginTop": 48,
                    "width": "100%",
                  },
                ]
              }
              width="100%"
            >
              <View
                style={
                  Array [
                    Object {
                      "width": 710,
                    },
                  ]
                }
                width={710}
              >
                <View
                  borderBottomColor="neutroFrio2"
                  borderBottomWidth="hairline"
                  flexDirection="row"
                  justifyContent="space-between"
                  style={
                    Array [
                      Object {
                        "borderBottomColor": "#8A8C8E",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      Object {
                        "overflow": "hidden",
                      },
                    ]
                  }
                >
                  <View
                    flexGrow={4}
                    style={
                      Array [
                        Object {
                          "flexGrow": 4,
                        },
                      ]
                    }
                  >
                    <TextInput
                      accessibilityLabel="com.usereserva:id/modal_sign_in_input_password"
                      allowFontScaling={true}
                      autoCapitalize="none"
                      autoCompleteType="off"
                      autoCorrect={false}
                      collapsable={false}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onGestureHandlerEvent={[Function]}
                      onGestureHandlerStateChange={[Function]}
                      placeholder="Digite sua senha"
                      rejectResponderTermination={true}
                      secureTextEntry={true}
                      style={
                        Object {
                          "margin": 0,
                          "maxWidth": 684,
                          "padding": 0,
                        }
                      }
                      testID="com.usereserva:id/modal_sign_in_input_password"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                  <View
                    justifyContent="center"
                    mr="xxxs"
                    style={
                      Array [
                        Object {
                          "justifyContent": "center",
                          "marginRight": 16,
                        },
                        Object {
                          "width": 40,
                        },
                      ]
                    }
                  >
                    <View
                      alignItems="center"
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "opacity": 1,
                          }
                        }
                        testID="com.usereserva:id/modal_sign_in_input_password_button_hide_password"
                      >
                        <RNSVGSvgView
                          bbHeight={22}
                          bbWidth={22}
                          color="preto"
                          focusable={false}
                          size={22}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              Array [
                                Object {
                                  "color": "#333333",
                                  "height": 22,
                                  "width": 22,
                                },
                              ],
                              Object {
                                "flex": 0,
                                "height": 22,
                                "width": 22,
                              },
                            ]
                          }
                        >
                          <RNSVGGroup>
                            <RNSVGSvgView
                              align="xMidYMid"
                              bbHeight="100%"
                              bbWidth="100%"
                              fill="none"
                              focusable={false}
                              height="100%"
                              meetOrSlice={0}
                              minX={0}
                              minY={0}
                              stroke="#333333"
                              strokeLinecap="round"
                              strokeLinejoin="round"
                              strokeWidth={2}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderWidth": 0,
                                  },
                                  Array [
                                    Object {
                                      "height": "100%",
                                      "width": "100%",
                                    },
                                  ],
                                  Object {
                                    "flex": 0,
                                    "height": "100%",
                                    "width": "100%",
                                  },
                                ]
                              }
                              vbHeight={24}
                              vbWidth={24}
                              width="100%"
                            >
                              <RNSVGGroup
                                fill={null}
                                propList={
                                  Array [
                                    "fill",
                                    "stroke",
                                    "strokeWidth",
                                    "strokeLinecap",
                                    "strokeLinejoin",
                                  ]
                                }
                                stroke={4281545523}
                                strokeLinecap={1}
                                strokeLinejoin={1}
                                strokeWidth={2}
                              >
                                <RNSVGPath
                                  d="M17.94 17.94A10.07 10.07 0 0112 20c-7 0-11-8-11-8a18.45 18.45 0 015.06-5.94M9.9 4.24A9.12 9.12 0 0112 4c7 0 11 8 11 8a18.5 18.5 0 01-2.16 3.19m-6.72-1.07a3 3 0 11-4.24-4.24"
                                />
                                <RNSVGPath
                                  d="M1 1L23 23"
                                />
                              </RNSVGGroup>
                            </RNSVGSvgView>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                mb="quarck"
                mt="micro"
                style={
                  Array [
                    Object {
                      "marginBottom": 4,
                      "marginTop": 12,
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="com.usereserva:id/modal_sign_in_cta_forgot_password"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="com.usereserva:id/modal_sign_in_cta_forgot_password"
                >
                  <Text
                    fontSize={14}
                    style={
                      Array [
                        Object {
                          "fontSize": 14,
                        },
                        Object {
                          "textDecorationLine": "underline",
                        },
                      ]
                    }
                  >
                    Esqueci minha senha
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            alignItems="stretch"
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "color": "#FFF",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 13,
                  "justifyContent": "center",
                  "marginTop": 32,
                  "minWidth": 130,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 16,
                }
              }
              testID="com.usereserva:id/modal_sign_in-cta_enter"
            >
              <View
                alignItems="center"
                flexDirection="row"
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <View
                  justifyContent="center"
                  style={
                    Array [
                      Object {
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    color="white"
                    fontFamily="nunitoBold"
                    fontSize={13}
                    letterSpacing={1.6}
                    style={
                      Array [
                        Object {
                          "color": "#FFF",
                          "fontFamily": "Nunito-Bold",
                          "fontSize": 13,
                          "letterSpacing": 1.6,
                        },
                      ]
                    }
                  >
                    CONTINUAR COMPRANDO
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            alignItems="center"
            flexDirection="row"
            justifyContent="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 24,
                  "marginTop": 24,
                },
              ]
            }
          >
            <View
              borderColor="divider"
              borderWidth="hairline"
              flex={1}
              marginRight="md"
              style={
                Array [
                  Object {
                    "borderColor": "#D8D9DA",
                    "borderWidth": 1,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginRight": 48,
                  },
                ]
              }
            />
            <Text
              fontSize={14}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                ]
              }
              textAlign="center"
            >
              Ainda não é
               
              <Text
                fontFamily="reservaSerifBlack"
                fontSize={14}
                style={
                  Array [
                    Object {
                      "fontFamily": "ReservaSerif-Black",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Prime
              </Text>
              ?
            </Text>
            <View
              borderColor="divider"
              borderWidth="hairline"
              flex={1}
              marginLeft="md"
              style={
                Array [
                  Object {
                    "borderColor": "#D8D9DA",
                    "borderWidth": 1,
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginLeft": 48,
                  },
                ]
              }
            />
          </View>
          <Text
            color="neutroFrio2"
            fontFamily="reservaSansRegular"
            fontSize={14}
            style={
              Array [
                Object {
                  "color": "#8A8C8E",
                  "fontFamily": "ReservaSans-Regular",
                  "fontSize": 14,
                },
                Object {
                  "marginBottom": 24,
                },
              ]
            }
          >
            Por apenas
             
            <Text
              color="fullBlack"
              fontFamily="reservaSansBold"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "ReservaSans-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              12x R$ 25
            </Text>
            , assinantes
             
            <Text
              color="fullBlack"
              fontFamily="reservaSerifBlack"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "ReservaSerif-Black",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Prime
            </Text>
             
            têm acesso a vantagens como descontos exclusivos e frete grátis em compras acima de
             
            <Text
              color="fullBlack"
              fontFamily="reservaSansBold"
              fontSize={14}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "ReservaSans-Bold",
                    "fontSize": 14,
                  },
                ]
              }
            >
              R$ 499
            </Text>
            , toque no botão abaixo e comece agora mesmo a comprar!
          </Text>
          <View
            alignItems="center"
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/modal_sign_in_cta_add_prime"
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": "100%",
                }
              }
              testID="com.usereserva:id/modal_sign_in_cta_add_prime"
            >
              <View
                alignItems="center"
                bg="verdeSucesso"
                height={48}
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#38A238",
                      "height": 48,
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              >
                <Text
                  color="white"
                  fontFamily="reservaSansMedium"
                  fontSize={14}
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontFamily": "ReservaSans-Medium",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  ADICIONAR PRIME AO CARRINHO
                </Text>
              </View>
            </View>
          </View>
          <Text
            color="neutroFrio2"
            fontSize={12}
            style={
              Array [
                Object {
                  "color": "#8A8C8E",
                  "fontSize": 12,
                  "marginTop": 12,
                },
              ]
            }
          >
            Para mais informações sobre o prime,
             
            <Text
              accessibilityLabel="com.usereserva:id/modal_sign_click_here"
              color="fullBlack"
              fontFamily="reservaSansBold"
              fontSize={12}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "ReservaSans-Bold",
                    "fontSize": 12,
                    "textDecorationColor": "black",
                    "textDecorationLine": "underline",
                    "textDecorationStyle": "solid",
                  },
                ]
              }
              testID="com.usereserva:id/modal_sign_click_here"
            >
              clique aqui
            </Text>
             
            antes de concluir seu pedido.
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
